pkg/scheduler/core/util.go

```diff
@@ -68,20 +69,33 @@ func calAvailableReplicas(clusters []*clusterv1alpha1.Cluster, spec *workv1alpha
 		return availableTargetClusters
 	}
 
-	// Get the minimum value of MaxAvailableReplicas in terms of all estimators.
+	// get estimators and sort it by priority.
 	estimators := estimatorclient.GetReplicaEstimators()
+	sort.Slice(estimators, func(i, j int) bool {
+		return estimators[i].GetPriority() > estimators[j].GetPriority()
+	})
+
+	priorityOfAvailableEstimator := estimatorclient.EstimatorPriority(math.MinInt32)
 	ctx := context.WithValue(context.TODO(), util.ContextKeyObject,
 		fmt.Sprintf("kind=%s, name=%s/%s", spec.Resource.Kind, spec.Resource.Namespace, spec.Resource.Name))
+
 	for _, estimator := range estimators {
+		// if higher-priority estimators have formed a full result of member clusters, no longer to call lower-priority estimator.
+		if estimator.GetPriority() < priorityOfAvailableEstimator && clustersFullyEstimated(availableTargetClusters) {
```

Comment:
Does that mean we have estimators with the same priority?
Otherwise, we don't need to compare the priority in range loops as the priority in different loops must have different priorities.